### PR TITLE
Made changes for PNP with Sanity and unit test

### PR DESCRIPTION
**Pull request by abimishr**

### CODE REVIEW

File: plugins/modules/pnp_intent.py [hunk 1]
```diff
@@ -609,86 +609,61 @@
...
-            pnp_type=dict(required=False, type=str, default="Default")
+        """Validate the fields provided in the playbook"""
+
+        if not self.config:
+            self.msg = "config not available in playbook for validation"
+            self.status = "success"
+            return self
+
+        pnp_spec = {
+            'template_name': {'required': True, 'type': 'str'},
```
Comment (madhansansel) [inline plugins/modules/pnp_intent.py:633]: 'template_name': {'type': 'str', 'required': True},

File: plugins/modules/pnp_intent.py [hunk 2]
```diff
@@ -804,73 +783,77 @@ def get_have(self):
...
-                project_name=params.get("project_name"),
-                template_name=params.get("template_name")
-            )
-
+        return self
+
+    def get_want(self, config):
+        """Get all the image, site and pnp related information from playbook that is needed to be created in DNAC"""
+
+        want = {
```
Comment (madhansansel) [inline plugins/modules/pnp_intent.py:815]: Can we write self.want = { }

**Review by madhansansel — APPROVED** [review]

File: plugins/modules/pnp_intent.py (hunk 1 above)
Comment (abimishr) [reply plugins/modules/pnp_intent.py]: This is done

File: plugins/modules/pnp_intent.py (hunk 2 above)
Comment (abimishr) [reply plugins/modules/pnp_intent.py]: This is done as well

**Review by madhansansel — APPROVED** [review]

**Review by madhansansel — APPROVED** [review]

**Pull request merged**